File: code/data_structures/arrays/search_element.cpp
#include<iostream>
using namespace std;
int main(){
    int arr[100],i,j,k,l,n,temp,flag=0;
    cout<<"Enter the size of the array: "<<endl;
    cin>>n;
    cout<<"Enter the elements"<<endl;
    for(i=0;i<n;i++){
        cin>>arr[i];
    }
    for(i=0;i<n;i++){
        for(j=i+1;j<n;j++){
            if(arr[j]<arr[i]){
                temp=arr[i];
                arr[i]=arr[j];
                arr[j]=temp;
            }
        }
    }
    cout<<"Sorted array is"<<endl;
    for(i=0;i<n;i++){
        cout<<arr[i]<<" ";
    }
    cout<<endl<<"enter the element you want to search"<<endl;
    cin>>l;
    int a=0,b=n,mid;
    while(a<=b){
        mid=(a+b)/2;
        if(arr[mid]==l){
            cout<<"Element found at index "<<mid<<endl;
            flag=1;
            break;
        }
        else if(arr[mid]>l)
            b=mid-1;
        else
            a=mid+1;

    }
    if(flag==0)
        cout<<"Element you are searching for is not in the array"<<endl;


}

File: code/data_structures/arrays/search_element.java
import java.util.Scanner;

class search_element{
	public static void main(String []args){
		Scanner obj=new Scanner(System.in);
		System.out.println("Enter the no of elememts");
		int n=obj.nextInt();
		int []a=new int[100];
		for(int i=0;i<n;i++){
			a[i]=obj.nextInt();
		}
		System.out.println("Enter the elememts");
 		for(int i=0;i<n;i++){
			for(int j=i+1;j<n;j++){
				if(a[i]>a[j]){
					int temp=a[i];
					a[i]=a[j];
					a[j]=temp;
				}
			}
		}
		System.out.println("The sorted array is ");
		for(int i=0;i<n;i++){
			System.out.println(a[i]);
		}
		System.out.println("Enter the element u want to search");
		int k=obj.nextInt();
		int p,flag=0,q,mid;
		p=0;
		q=n;
		
		while(p<=q){
			mid=(p+q)/2;
			if(a[mid]==k){
				flag=1;
				System.out.println("Element found at index "+mid);
				break;
			}
			else if(k>a[mid]){
				p=mid+1;			
			}
			else{
				q=mid-1;
			}
		}		
		if(flag==0){
			System.out.println("The element you are searching for doesn't exist in the array");
		}		

	}




}
